spec/models/user_spec.rb

```diff
@@ -0,0 +1,191 @@
+require 'rails_helper'
+                                                                #UserModelUserModelUserModel
+describe User, type: :model do
+  let(:user) { create(:user) }
+
+  it 'is expected to have a valid factory' do
+    expect(user.valid?).to eq true
+  end
+
+  it { should have_many(:articles) }
+
+  it { should validate_presence_of(:email) }
+  it { should validate_presence_of(:nickname) }
+  it { should validate_presence_of(:nickname) }
+
+  it { should validate_uniqueness_of(:nickname).case_insensitive }
+
+  #it { should_validate_length_of (:nickname).is_at_most(20) }
+end
+                                                              #ArticleModelArticleModel
+describe Article, type: :model do
+
+  it { should belong_to(:user) }
+  it { should belong_to(:category) }
+
+  it { should validate_presence_of(:title) }
+  it { should validate_presence_of(:content) }
+  it { should validate_presence_of(:user_id) }
+#  it { should validate_presence_of(:category_id) }
+
+  it { should validate_uniqueness_of(:title).case_insensitive }
+
+end
+
+                                                              #CategoryModelCategoryModel
+  describe Category, type: :model do
+   it { should have_many(:articles) }
+   it { should validate_presence_of(:name) }
+  end
+
+                                                                    #UserController#UserController
+describe UsersController, type: :controller do
+ it { should use_before_action(:authenticate_user!) }
+end
+
+                                                                      #FeatureFeature
+ describe 'succesfull user creating', type: :feature do
+   it 'should add new user to database' do
+     u1 = User.all.count
+     visit new_user_registration_path
+     fill_in 'Nickname', with: 'asdfgh'
+     fill_in 'Email', with: 'asdfgh@example.com'
+     fill_in 'Password', with: 'qwerty'
+     fill_in 'Password confirmation', with: 'qwerty'
+     click_button 'Sign up'
+     u2 = User.all.count
+     expect(u2 - u1).to eq(1)
```

Comment:
expect do
     visit new_user_registration_path
     fill_in 'Nickname', with: 'asdfgh'
     fill_in 'Email', with: 'asdfgh@example.com'
     fill_in 'Password', with: 'qwerty'
     fill_in 'Password confirmation', with: 'qwerty'
     click_button 'Sign up'
end.to change { User.count }.by(1)

Reply:
+++